crud/select/filters.lua

```diff
@@ -163,10 +164,13 @@ local function format_value(value)
         return ("%q"):format(value)
     elseif type(value) == 'number' then
         return tostring(value)
-    elseif type(value) == 'cdata' then
-        return tostring(value)
     elseif type(value) == 'boolean' then
         return tostring(value)
+    elseif type(value) == 'cdata' then
```

Comment:
Seems that `cdata` check is redundant.

Reply:
`cdata` was here before, removing it will change the type checking logic.

Reply:
No, cdata is needed e.g. for "decimals" and cdata numbers

```
tarantool> type(1ULL)
---
- cdata
...
```

Reply:
It seems that my comment wasn't correct.

I mean that UUID check can be performed without checking cdata before.

```lua
elseif utils.is_uuid(value) then
...
elseif type(value) == 'cdata' then
...
```

Reply:
I don't agree. It looks clear and uses the desired field type, not the runtime type of the value.
Btw, checking for uuid type requires checking for cdata too:

```
tarantool> ffi = require'ffi'
---
...

tarantool> ffi.istype(nil, 'uuid')
---
- error: 'bad argument #1 to ''?'' (C type expected, got nil)'
...

tarantool> type(nil)
---
- nil
...

tarantool> ffi.istype(1, 'uuid')
---
- error: 'bad argument #1 to ''?'' (C type expected, got number)'
...

tarantool> type(1)
---
- number
...
```

Feel free to refactor this place in further patches.

Reply:
The checks in the comment above are wrong, (wrong order of arguments):

```
tarantool> ffi = require'ffi'
---
...

tarantool> local uuid_t = ffi.typeof('struct tt_uuid')
---
...

tarantool> uuid_t = ffi.typeof('struct tt_uuid')
---
...

tarantool> ffi.istype(uuid_t, nil)
---
- false
...

tarantool> ffi.istype(uuid_t, 1)
---
- false
...

tarantool> ffi.istype(uuid_t, "value")
---
- false
...

tarantool> ffi.istype(uuid_t, box.NULL)
---
- false
...
```

Moved the check as suggested.